### PR TITLE
update pipeline register error check to pass pipelines registered at runtime

### PR DESCRIPTION
**Pull request by bfineran**
currently supported tasks check-register will fail for tasks that are not included in `SupportedTasks`. That class is only meant to deal with predefined NM pipelines - this PR adds a layer to allow for tasks registered at runtime

**test_plan:**
manual - verified unblock of custom pipeline tutorial

### CODE REVIEW

File: src/deepsparse/tasks.py [hunk 1]
```diff
@@ -137,10 +144,11 @@ def check_register_task(cls, task: str):
             # register with Pipeline.register
             import deepsparse.transformers.haystack  # noqa: F401
 
-        else:
+        all_tasks = set(cls.task_names() + (list(extra_tasks or [])))
```
Comment (rahul-tuli) [inline src/deepsparse/tasks.py:147]:
```suggestion
        all_tasks = set(cls.task_names() + list(extra_tasks or []))
```

File: src/deepsparse/tasks.py [hunk 2]
```diff
@@ -110,7 +110,14 @@ class SupportedTasks:
     all_task_categories = [nlp, image_classification, yolo, yolact, haystack]
 
     @classmethod
-    def check_register_task(cls, task: str):
+    def check_register_task(
+        cls, task: str, extra_tasks: Optional[Iterable[str]] = None
```
Comment (rahul-tuli) [inline src/deepsparse/tasks.py:114]: Nice to not force extra tasks to be a list.

**Review by rahul-tuli — APPROVED** [review]

**Review by rahul-tuli — COMMENTED** [review]
Reviewed this by mistake

**Review by KSGulin — APPROVED** [review]
Some tests are failing. Otherwise LGTM

**Pull request merged**